### ISSUE TITLE
Add ability to use DOM elements as popover content

### ISSUE THREAD

**Issue by Jezza** [twbs/bootstrap#20219]
There were a couple of good reasons for implementing this, and they were brought up with the initial issue: #3722.

I'll repeat them here for completionist sake though.
- Event bindings break if you copy the html, and not use the actual element.
- This would let the popover work nicer with other JS frameworks, such as AngularJS.

_Current workaround_:
Just copy the html, and rebind the events. It's not fun or pretty.

_Side note:_
The initial issue (#3722) was closed simply because of lack of action, and the final comment: "Locking this due to its age. Added a note to look at this for v4 though." , which was stupid.
You don't close a ticket because of it's age, let alone if the ticket might be considered for a later version.
Keeping it open means people don't have to go through 13,000 other tickets to check if any are still actively wanted/needed.
Otherwise you run the risk of it just vanishing into the ether.

So, closing it was a stupid idea, and brings to question how many other issues have been closed similarly, and then completely forgotten?
In short, don't close a ticket like that.

**Comment by alecpl** [comment]
Another issue here is that the content node is out of DOM tree of the page after you use `.popover('hide')`. So, it is not really possible to simply work with the popup content when it is in hidden state. And that's a real pita in my use-case.
Any workarounds? How about appending the content node back to the element specified in 'container' option (default document.body) and making it invisible on hide event?

**Comment by alecpl** [comment]
@FezVrasta Maybe you have some hints for my issue? Not sure it's popper or bootstrap implementation limitation.

**Comment by FezVrasta** [comment]
I don't think I understand where the problem is.

**Comment by alecpl** [comment]
@FezVrasta At least with current Bootstrap implementation when popover is being shown the DOM element is appended to the popover. When the popover is hidden it's no longer in DOM tree of the page, so I can't access it with e.g. jquery css selectors.

**Comment by FezVrasta** [comment]
Couldn't Bootstrap simply hide it visually and not removing it from DOM in case you used an already existing DOM node as content?

**Comment by alecpl** [comment]
That's a question, yeah. @Johann-S ideas?

**Comment by Johann-S** [comment]
What's @FezVrasta can be a possibility yes.
I remember they are some features asked about having a possibility to update the contents of a Popover

**Comment by alecpl** [comment]
Thanks. I don't feel confident to digg into Bootstrap code. For now I workaround this by:
```
$(button).on('hide.bs.popover', function() {
        $(popup).appendTo(document.body);
})
```

**Comment by mdo** [comment]
Closing as stale. Happy to revisit if someone wants to help us take this on.
